assets/pages/regular-contributions/components/emailFormField.jsx

```diff
@@ -0,0 +1,68 @@
+// @flow
+
+// ----- Imports ----- //
+
+import React from 'react';
+import { connect } from 'react-redux';
+import type { Dispatch } from 'redux';
+
+import TextInput from 'components/textInput/textInput';
+import { setEmail, type Action } from 'helpers/user/userActions';
+
+
+// ----- Types ----- //
+
+type PropTypes = {
+  emailUpdate: (email: string) => void,
+  email: string,
+  isSignedIn: boolean,
+};
+
+
+// ----- Component ----- //
+
+const EmailFormField = (props: PropTypes) => {
+
+  if (props.isSignedIn) {
+    return null;
+  }
+
+  return (<TextInput
+    id="email"
+    value={props.email}
+    labelText="Email"
+    placeholder="Email"
+    onChange={props.emailUpdate}
+    modifierClasses={['email']}
+    required
+  />);
+
+};
+
+
+// ----- Map State/Props ----- //
+
+function mapStateToProps(state) {
+  const { user } = state.page;
+  return {
+    email: user.email,
+    isoCountry: state.common.internationalisation.countryId,
+    isSignedIn: state.page.user.isSignedIn,
+  };
+
+}
+
+function mapDispatchToProps(dispatch: Dispatch<Action>) {
+
+  return {
+    emailUpdate: (email: string) => {
+      dispatch(setEmail(email));
+    },
+  };
+
+}
+
+
+// ----- Exports ----- //
+
+export default connect(mapStateToProps, mapDispatchToProps)(EmailFormField);
```

Comment:
This is a clone of `one-off/emailFormField.jsx`? Would it be preferable to create a shared component and provide containers for the two pages? So:

```
components/emailFormField/emailFormField.jsx
pages/oneoff-contributions/components/emailFormFieldContainer.js
pages/regular-contributions/components/emailFormFieldContainer.js
```

with the containers looking something like this:

```js
// ----- Imports ----- //

import { connect } from 'react-redux';

import { setEmail, type Action } from 'helpers/user/userActions';
import EmailFormField from 'components/emailFormField/emailFormField';

import type { State } from '../oneOffContributionsReducer';


// ----- State/Action Maps ----- //

function mapStateToProps(state: State) {

  return {
    email: state.page.user.email,
    isoCountry: state.common.internationalisation.countryId,
    isSignedIn: state.page.user.isSignedIn,
  };

}

const mapDispatchToProps = { emailUpdate: setEmail };


// ----- Exports ----- //

export default connect(mapStateToProps, mapDispatchToProps)(EmailFormField);
```

**Side Note**: Doesn't look like `isoCountry` is a prop of this component any more, worth dropping it from the state mapping?